Improve navigation flow and implement navigation breadcrumbs

praseodym opened the pull request:
Resolves #214.

Lionqueen94 commented on frontend/lib/ui/style/layout.css:
```diff
@@ -17,13 +17,18 @@
 
   > nav {
     flex: 0 0 44px;
+    gap: 8px;
```
```suggestion
    gap: 0.5rem;
```

Lionqueen94 commented on frontend/lib/ui/style/layout.css:
```diff
@@ -35,6 +40,10 @@
       border-bottom: 4px solid var(--accent-orange);
     }
+    svg {
+      width: 24px;
```
```suggestion
      width: 1.5rem;
```

Lionqueen94 commented on frontend/lib/ui/style/table.css:
```diff
@@ -2,6 +2,7 @@ table.overview_table {
   width: 100%;
   line-height: 1.5rem;
   border-spacing: unset;
+  color: var(--gray-600);
```
```suggestion
  color: var(--color-help-text);
```

gray-600 will be renamed in the design fixes PR

praseodym commented:
@jorisleker can you confirm that this resolves everything in #214?

jorisleker reviewed (changes requested):
@praseodym to save space, we do not show the current page in the breadcrumb. That means that on `/1/input` we do not show the name of the elections, and on `/1/input/1/..` the name of the pollingstation is not shown. Only exception /overview itself, there we show 'Overzicht'.  

![Scherm­afbeelding 2024-08-14 om 08 01 27](https://github.com/user-attachments/assets/283adf99-f8c4-4982-ba5c-86f30b32ec63)

jschuurk-kr commented on frontend/app/module/input/PollingStation/PollingStationLayout.test.tsx:
```diff
@@ -41,7 +41,7 @@ describe("PollingStationLayout", () => {
       </ElectionListProvider>,
     );
-    expect(await screen.findByText(pollingStationMock.name));
+    expect(await screen.findAllByText(pollingStationMock.name));
```
`findAllByText()` is too broad. We should use [`within()`](https://testing-library.com/docs/dom-testing-library/api-within/) to do separate asserts for the navbar and the header.

When it's a link, we should use `getByRole('link', { name: '<text>' })`. The more specific we can make the locators (without getting into testing-the-implementation territory), the better our test coverage.

For the header we should switch to `screen.getByRole("heading", { level: 1, name: '<text>' })`.

The same thing needs to happen in `InputHomePage.test.tsx`.

praseodym replied:
Improved in ac58fd5.

jschuurk-kr reviewed (changes requested):
I can't get the dialog that we show when the user clicks 'Invoer afbreken' to show up when I use one of the navigation links.

I'm missing tests in which the navigation is used.

praseodym commented:
> I can't get the dialog that we show when the user clicks 'Invoer afbreken' to show up when I use one of the navigation links.
> 
> 
> 
> I'm missing tests in which the navigation is used.

This is not implemented in this PR because navigation code is only implemented in #133.

praseodym commented:
> I can't get the dialog that we show when the user clicks 'Invoer afbreken' to show up when I use one of the navigation links.

This is not implemented in this PR because navigation code is only implemented in #133. 

I've changed this PR not to close #214 so that we can implement that part once #133 and #212 are implemented. 

> I'm missing tests in which the navigation is used.

What would be useful to test about the currently implemented navigation? Test that the links work?

jschuurk-kr commented:
Thanks for the scope clarification!

> > I'm missing tests in which the navigation is used.
> 
> What would be useful to test about the currently implemented navigation? Test that the links work?

Yes. Which might seem a bit much right now, but is likely to be helpful some years down the line.

On the other hand, soon enough we'll have the dialog and those tests will need to be changed, so "some years down the line" is not going to happen. So fine with me to not include these tests here and wait until all parts of the navigation come together.

praseodym commented:
> @praseodym to save space, we do not show the current page in the breadcrumb. That means that on `/1/input` we do not show the name of the elections, and on `/1/input/1/..` the name of the pollingstation is not shown. Only exception /overview itself, there we show 'Overzicht'.

Done in 40a32ec. I initially used the wrong design from the components section in Figma.

> #214 also request to show the name of the logged in user in the top right corner of the navbar. Can we put a placeholder there?

I've added a placeholder with name "Invoerder" in 3a1ba26. Even though we will display an actual user's full name later, I think adding a fake name now will be confusing. I'm open to other suggestions.

> > What would be useful to test about the currently implemented navigation? Test that the links work?
> 
> Yes. Which might seem a bit much right now, but is likely to be helpful some years down the line.

I've added tests in ac58fd5.

lkleuver commented on frontend/app/component/navbar/NavBar.tsx:
```diff
@@ -0,0 +1,15 @@
+
+import { IconUser } from "@kiesraad/icon";
+
+export const NavBar = ({ children }: { children: React.ReactNode }) => {
```
Personally prefer function over arrow function like the other components.

praseodym replied:
Changed in 105677be5f84a8008c4926b8a939fdf0e0a2ca62.

lkleuver commented on frontend/app/component/navbar/NavBar.tsx:
```diff
@@ -0,0 +1,15 @@
+export const NavBar = ({ children }: { children: React.ReactNode }) => {
+  return (
+    <nav aria-label="primary-navigation">
+      <div className="links">{children}</div>
```
especially with a classname like "links" it might be better to use css modules.

lkleuver commented on frontend/lib/icon/gen.tsx:
```diff
@@ -131,6 +131,20 @@ export const IconThumbsUp = () => (
 );
+export const IconUser = () => (
+  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <g id="user-01">
```
We can't use ID attributes in icons unless we ever only want to render one per page.

praseodym replied:
Fixed in 9b17f82535ee7e629a5aa6ca929e136f307c8bce.

lkleuver commented on frontend/lib/icon/gen.tsx:
```diff
@@ -131,6 +131,20 @@ export const IconThumbsUp = () => (
+  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <g id="user-01">
+      <path
+        id="Icon"
```
idem

praseodym replied:
Fixed in 9b17f82535ee7e629a5aa6ca929e136f307c8bce.

jschuurk-kr commented:
> @praseodym to save space, we do not show the current page in the breadcrumb. That means that on `/1/input` we do not show the name of the elections, and on `/1/input/1/..` the name of the pollingstation is not shown. Only exception /overview itself, there we show 'Overzicht'.

@jorisleker I find this an odd experience, esp. the way "Overzicht" changes from text with an orange underline on the Overview page to a link on the Input page. Curious to see if this comes up in the user testing or not.

Secondly, does it make sense to change "Overzicht" to "Verkiezingen"? "Overzicht" makes me wonder "Overview of what?"

jorisleker commented:
> @jorisleker I find this an odd experience, esp. the way "Overzicht" changes from text with an orange underline on the Overview page to a link on the Input page. Curious to see if this comes up in the user testing or not.

Agreed. The design with underline is meant for when we have more menu items (for coordinator/beheerder roles). For invoerder this is not the case. Design should be same as on the `/1/input`page

jorisleker commented:
Not sure if this is related to this issue, but when I click the 'overzicht' item in the breadcrumb, I sometimes get a page that shows the breadcrumb, but the rest op the page remains blank. Got this console error: 
 
![Scherm­afbeelding 2024-08-15 om 12 12 33](https://github.com/user-attachments/assets/265dde52-76be-4b12-950a-0d874a474c03)

Seems to happen when I've been 'away' for a while. Maybe when session has expired?

lkleuver commented:
> Not sure if this is related to this issue, but when I click the 'overzicht' item in the breadcrumb, I sometimes get a page that shows the breadcrumb, but the rest op the page remains blank. Got this console error:
> 
> ![Scherm­afbeelding 2024-08-15 om 12 12 33](https://private-user-images.githubusercontent.com/3286523/358191353-265dde52-76be-4b12-950a-0d874a474c03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MjIwMDksIm5iZiI6MTcyMzcyMTcwOSwicGF0aCI6Ii8zMjg2NTIzLzM1ODE5MTM1My0yNjVkZGU1Mi03NmJlLTRiMTItOTUwYS0wZDg3NGE0NzRjMDMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTVUMTEzNTA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2RmZjU5YmY3ZDhlOTAyNzlhNzBlMDMyYzFlMWUxOTdiNjA0MTQyOThhYmVlYWE3MDE2MzkxZTNmOTliNmI3NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ciQTL9vml9zChLTz92mKdP6gHcWrMMe8fSFJXzwK488)
> 
> Seems to happen when I've been 'away' for a while. Maybe when session has expired?

I think this is when the apiclient tries to JSON parse a non json response.

praseodym commented:
> > @jorisleker I find this an odd experience, esp. the way "Overzicht" changes from text with an orange underline on the Overview page to a link on the Input page. Curious to see if this comes up in the user testing or not.
> 
> Agreed. The design with underline is meant for when we have more menu items (for coordinator/beheerder roles). For invoerder this is not the case. Design should be same as on the `/1/input`page

Does that mean we'll have 'Overzicht' as a link on the `/1/input` page, but 'Overzicht' with the same style but no link on the `/` page? That seems even more inconsistent.

Also, in the future, can we try to discuss the design before it is implemented?

jorisleker commented:
> Does that mean we'll have 'Overzicht' as a link on the /1/input page, but 'Overzicht' with the same style but no link on the / page? That seems even more inconsistent.

That's what it means. Alternative is to leave it out all together on the overview page. Fine with me too

> Also, in the future, can we try to discuss the design before it is implemented?
Yes. That was the goal, ticket was assigned to me to finish design before implementation :-)

praseodym commented:
> > Does that mean we'll have 'Overzicht' as a link on the /1/input page, but 'Overzicht' with the same style but no link on the / page? That seems even more inconsistent.
> 
> That's what it means. Alternative is to leave it out all together on the overview page. Fine with me too

Changed in 2c094f421a4fbe8186f5edf68627403b9729ec79.
>
> > Also, in the future, can we try to discuss the design before it is implemented?
> 
> Yes. That was the goal, ticket was assigned to me to finish design before implementation :-)

I see! My understanding was that the design was already finished.

praseodym commented:
Fixed merge conflicts.

praseodym commented:
> > I can't get the dialog that we show when the user clicks 'Invoer afbreken' to show up when I use one of the navigation links.
> 
> This is not implemented in this PR because navigation code is only implemented in #133.
> 
> I've changed this PR not to close #214 so that we can implement that part once #133 and #212 are implemented.

I've created #265 to implement the blocking UI, because this is specific to the data entry screens and not navigation in general.

praseodym commented:
Fixed new merge conflicts.
